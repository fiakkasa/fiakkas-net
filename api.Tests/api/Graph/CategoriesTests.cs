using HotChocolate.Execution;

namespace api.Tests;

public class CategoriesTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task Categories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  categories {
    totalCount
    items {
      __typename
      createdAt
      id
      title
      updatedAt
      version
      ... on InformationTechnologyCategory {
        createdAt
        href
        id
        title
        updatedAt
        version
        portfolioCategories {
          totalCount
        }
        portfolioCustomers {
          totalCount
        }
        portfolioItems {
          totalCount
        }
      }
      ... on OtherCategory {
        createdAt
        id
        title
        updatedAt
        version
      }
      ... on PortfolioCategory {
        createdAt
        id
        title
        updatedAt
        version
        customers {
          totalCount
        }
        portfolioItems {
          totalCount
        }
        technologyCategories {
          totalCount
        }
      }
      ... on ResumeCategory {
        associatedCategoryTypes
        createdAt
        id
        title
        updatedAt
        version
        associatedCategories {
          totalCount
        }
        educationItems {
          totalCount
        }
      }
      ... on SoftwareDevelopmentCategory {
        createdAt
        href
        id
        title
        updatedAt
        version
        portfolioCategories {
          totalCount
        }
        portfolioCustomers {
          totalCount
        }
        portfolioItems {
          totalCount
        }
      }
      ... on UnknownCategory {
        createdAt
        id
        title
        updatedAt
        version
      }
      ... on UnknownTechnologyCategory {
        createdAt
        href
        id
        title
        updatedAt
        version
        portfolioCategories {
          totalCount
        }
        portfolioCustomers {
          totalCount
        }
        portfolioItems {
          totalCount
        }
      }
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task PortfolioCategories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  portfolioCategories {
    totalCount
    items {
      createdAt
      id
      title
      updatedAt
      version
      customers {
        totalCount
        items {
          createdAt
          href
          id
          title
          updatedAt
          version
        }
      }
      portfolioItems {
        totalCount
        items {
          categoryId
          createdAt
          customerId
          href
          id
          technologiesSummary
          technologyIds
          title
          updatedAt
          version
          year
        }
      }
      technologyCategories {
        totalCount
        items {
          createdAt
          href
          id
          title
          updatedAt
          version
        }
      }
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task ResumeCategories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  resumeCategories {
    totalCount
    items {
      associatedCategoryTypes
      createdAt
      id
      title
      updatedAt
      version
      associatedCategories {
        totalCount
        items {
          __typename
          createdAt
          id
          title
          updatedAt
          version
          ... on Category {
            createdAt
            id
            title
            updatedAt
            version
          }
          ... on InformationTechnologyCategory {
            createdAt
            href
            id
            title
            updatedAt
            version
          }
          ... on OtherCategory {
            createdAt
            id
            title
            updatedAt
            version
          }
          ... on PortfolioCategory {
            createdAt
            id
            title
            updatedAt
            version
          }
          ... on ResumeCategory {
            associatedCategoryTypes
            createdAt
            id
            title
            updatedAt
            version
          }
          ... on SoftwareDevelopmentCategory {
            createdAt
            href
            id
            title
            updatedAt
            version
          }
          ... on TechnologyCategory {
            createdAt
            href
            id
            title
            updatedAt
            version
          }
        }
      }
      educationItems {
        totalCount
        items {
          categoryId
          createdAt
          description
          href
          id
          location
          subjects
          title
          updatedAt
          version
        }
      }
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task InformationTechnologyCategories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  informationTechnologyCategories {
    totalCount
    items {
      createdAt
      href
      id
      title
      updatedAt
      version
      portfolioCategories {
        totalCount
        items {
          createdAt
          id
          title
          updatedAt
          version
        }
      }
      portfolioCustomers {
        totalCount
        items {
          createdAt
          href
          id
          title
          updatedAt
          version
        }
      }
      portfolioItems {
        totalCount
        items {
          categoryId
          createdAt
          customerId
          href
          id
          technologiesSummary
          technologyIds
          title
          updatedAt
          version
          year
        }
      }
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task SoftwareDevelopmentCategories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  softwareDevelopmentCategories {
    totalCount
    items {
      createdAt
      href
      id
      title
      updatedAt
      version
      portfolioCategories {
        totalCount
        items {
          createdAt
          id
          title
          updatedAt
          version
        }
      }
      portfolioCustomers {
        totalCount
        items {
          createdAt
          href
          id
          title
          updatedAt
          version
        }
      }
      portfolioItems {
        totalCount
        items {
          categoryId
          createdAt
          customerId
          href
          id
          technologiesSummary
          technologyIds
          title
          updatedAt
          version
          year
        }
      }
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task TechnologyCategories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  technologyCategories {
    totalCount
    items {
      __typename
      createdAt
      href
      id
      title
      updatedAt
      version
      ... on InformationTechnologyCategory {
        createdAt
        href
        id
        title
        updatedAt
        version
      }
      ... on SoftwareDevelopmentCategory {
        createdAt
        href
        id
        title
        updatedAt
        version
      }
      ... on UnknownTechnologyCategory {
        createdAt
        href
        id
        title
        updatedAt
        version
      }
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task OtherCategories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  otherCategories {
    totalCount
    items {
      createdAt
      id
      title
      updatedAt
      version
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
