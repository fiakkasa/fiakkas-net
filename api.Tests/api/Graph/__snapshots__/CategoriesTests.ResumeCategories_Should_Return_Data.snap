{
  "data": {
    "resumeCategories": {
      "totalCount": 1,
      "items": [
        {
          "associatedCategoryTypes": [
            "SOFTWARE_DEVELOPMENT"
          ],
          "createdAt": "2024-01-01T00:00:00.000Z",
          "id": "eb9d6258-99c4-46bd-bd44-23d35b19965d",
          "title": "Title",
          "updatedAt": null,
          "version": 1,
          "associatedCategories": {
            "totalCount": 1,
            "items": [
              {
                "__typename": "SoftwareDevelopmentCategory",
                "createdAt": "2024-01-01T00:00:00.000Z",
                "id": "ca832bf9-b7cb-4c31-bf8d-00f87a276fe3",
                "title": "Title",
                "updatedAt": null,
                "version": 1,
                "href": "/test"
              }
            ]
          },
          "educationItems": {
            "totalCount": 1,
            "items": [
              {
                "categoryId": "eb9d6258-99c4-46bd-bd44-23d35b19965d",
                "createdAt": "2024-01-01T00:00:00.000Z",
                "description": "Description",
                "href": "/test",
                "id": "38898c62-161e-40f2-8a9f-39bf1ff46224",
                "location": "Location",
                "subjects": [
                  "Subject"
                ],
                "title": "",
                "updatedAt": null,
                "version": 1
              }
            ]
          }
        }
      ]
    }
  }
}
